gapii/client/capture.go

```diff
@@ -193,21 +202,35 @@ func (p *Process) Capture(ctx context.Context, start task.Signal, stop task.Sign
 		}
 	}()
 
-	if p.conn == nil {
+	// TODO(rosasco): Pascal, why was this block conditionalized on p.Conn being nil ?
+	//                This assumes if p.Conn is initialized rather than the port, that the
+	//                header has already been sent.  Supposedly, it's ok to initialize either
+	//                p.Conn or p.Port prior to calling this Capture() method.
+	//                The Process struct is defined in "adb.go".
+	//                Is "p.Port == 0" a reasonable condition here (it is true on Fuchsia) ?
```

Comment:
Conclusion: Since you renamed the field and I don't see anywhere in this change where it used to be initialized, as long as everything builds, it means that we are safe and you can just remove the if.

I don't know. Maybe there used to be some codepath that initialized the process struct with a connection that is ready to trace? If that exists, I don't know if `p.Port == 0` is an ok check to ignore the existing connection and try to read the header.

Maybe the if was just defensive in an attempt to support a (future) case where the connection is already initialized - or a left over of a no longer supported past case. If so, you can just remove the if and "change" it so that it means that the connection is ready, but still want to read the header, which makes sense to me.